org.eclipse.scout.rt.mail/src/main/java/org/eclipse/scout/rt/mail/imap/ImapServerConfig.java

```diff
@@ -146,4 +149,18 @@ public ImapServerConfig withAdditionalSessionProperties(Map<String, String> addi
     m_additionalSessionProperties = additionalSessionProperties;
     return this;
   }
+
+  public Callable<String> getOAuthTokenResolver() {
+    return m_oauthTokenResolver;
+  }
+
+  /**
+   * The designated access token resolver to use OAuth2, if provided the <i>mail.imap.auth.mechanisms</i> property is
+   * set to <i>XOAUTH2</i> and the possibly provided password is overwritten with the access token. Set this property to
+   * <i>null</i> if no OAuth2 authentication should be used.
+   */
+  public ImapServerConfig withOauthTokenResolver(Callable<String> oauthTokenResolver) {
```

Comment:
Do we even need this resolver, why are additionalSessionProperties not used here?